.github/workflows/pytest.yml

```diff
@@ -0,0 +1,28 @@
+name: pytest
+
+on:
+    pull_request:
+    push:
+        branches: [main]
```

Comment:
Oh and also let's not restrict it to just main. Ideally we want the PR to run the tests also so we know if we can safely merge so I think you could just do:
```yaml
on: ['push','pull_request']
```

Reply:
This acutally runs on pull requests. this syntax just fixes a problem when making a PR.
With `on: [push, pull_request]` it will run twice on pull requests (once because it's a push, and once because it's a pull request).
With the one I have provided it will run on on all pull requests but only on push if it's the main branch.
I hope this is clear enough

Reply:
OH yeah yeah you're right. I'm not sure why it's not running the tests though? Is it because we have to merge it first?

or maybe because it's coming from a fork... Need to look, either way I think it's fine to merge as is then whenever you have the change for the requirements in :)

Reply:
Yeah. Probably not running the tests because they're workflows in a PR.